.github/workflows/check-pr.yml

```diff
@@ -1,30 +1,35 @@
 name: Check Pull Requests
 
+# Set the permissions of the github token to the minimum and only enable what is needed
+# See https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions
+permissions: {}
+
 on:
+  # this workflow is required to be run on pull_request_target as it modifies the PR comments
+  # care should be taken that the jobs do not run any untrusted input
+  # zizmor: ignore[dangerous-triggers]
```

Comment:
This is the only ignored rule - the threat is that running any external code in the context of the repo could make it possible for a PR to do malicious things. The jobs in this workflow avoid running any untrusted input.